deploy: scream for attention on certain changes

heerener opened the pull request.

heerener committed:
deploy: file a ticket if apptainer or singularity are changed

heerener committed:
Temporary change to make sure the ticket script works

heerener committed:
Typo fix and notifications dependency

Running too early is pointless

heerener committed:
Ability to file HELP tickets

heerener committed:
Disable ticket creation for now

heerener committed:
More info in the ticket

matz-e commented on notifications/apptainer_singularity.txt:
Will end up in `bluebrain/notifications` I hope?

heerener replied:
It did now :-)

heerener committed:
Interpret \n in echo and move notifications under bluebrain

heerener committed:
Make the pedantry happy

heerener committed:
Revert "Temporary change to make sure the ticket script works"

This reverts commit 00059600da320b7cec6b6c9f0f7bea72034fe9f5.

heerener committed:
File HELP tickets instead of BSD

matz-e commented on .gitlab-ci.yml:
```diff
@@ -201,3 +201,20 @@ clean_merged_prs:
+  needs: [atomic_build]
+  image: python:3.11-bullseye
+  rules:
+    - if: '$CI_COMMIT_BRANCH == $DEPLOYMENT_DEFAULT_BRANCH'
```
Sorry, I think it's not going to be _quite_ as simple as this.

If a dependency is changed, singularity would be rebuilt, too. That would require a trigger, too.

I'm wondering if we have something in the artifacts we could test for... :thinking:

matz-e commented on .gitlab-ci.yml:
```diff
@@ -201,3 +201,20 @@ clean_merged_prs:
+        - var/spack/repos/builtin/packages/singularityce/*
+  script:
+    - pip install -r bluebrain/notifications/requirements.txt
+    - git log -n 3 >> bluebrain/notifications/apptainer_singularity.txt
+    - echo -e '{noformat}\n\nThank you!' >> bluebrain/notifications/apptainer_singularity.txt
```
I'm wondering if we could use some formatting inside the template and add something like
```
{git_log}

Thank you!
```
there, then read the file and do `contents.format(git_log=subprocess.check_output(["git", ...])`?

Then it's all nicely contained in the Python script + template.

Would also allow to replace the git history beginning of the URL with some environment variable pointing to the Github repo

heerener committed:
Temp changes:

  * disable ticket creation
  * trigger on external pull requests
  * change singularityce

heerener committed:
Do the ticket description in the python script

heerener committed:
Search through the atomic_build test results

If there's an apptainer, singularity or singularityce test, we have to
file a ticket.

heerener committed:
Waahh, yaml has a colon problem

heerener committed:
We already have one of those

matz-e commented on .gitlab-ci.yml:
```diff
@@ -214,7 +214,21 @@ notifications:
+      # TODO: API_READ_TOKEN should be replaced by one not tied to my user account
+    - wget --header "Content-Type:application/json" --header "PRIVATE-TOKEN:${API_READ_TOKEN}" -O- "${CI_API_V4_URL}/projects/${CI_PROJECT_ID}/pipelines/${ATOMIC_BUILD_ID}/test_report"
+    - |
+      wget --header "Content-Type:application/json" --header "PRIVATE-TOKEN:${API_READ_TOKEN}" -O- "${CI_API_V4_URL}/projects/${CI_PROJECT_ID}/pipelines/${ATOMIC_BUILD_ID}/test_report" | jq -e '.test_suites[] | select(.name == "externals") .test_cases[] | select(.classname == "singularity" or .classname == "singularityce" or .classname == "apptainer")'
```
Final test: modify an irrelevant package and see if this gets triggered? I don't see anything that qualifies if the package should be a successful / skipped / failed test.

heerener replied:
I was under the impression that the test just had to be present, but you're right, skipped tests should be ignored. Added that now.

https://bbpgitlab.epfl.ch/hpc/personal/heeren/cicd/-/pipelines/141632 demonstrates the jq syntax, experiments were done here: https://bbpgitlab.epfl.ch/hpc/personal/heeren/cicd/-/blob/fetch-tests-through-api/.gitlab-ci.yml (feel free to re-run the pipeline on branch `fetch-tests-through-api`, it doesn't take long).

heerener committed:
Don't include skipped tests

heerener committed:
Switch to a non-relevant package

heerener committed:
Undo temporary changes

heerener commented:
https://bbpgitlab.epfl.ch/hpc/spack/-/pipelines/141634 modified neuron instead of singularityce; no notifications run was triggered.

All temporary changes have been undone again, we should be good to merge.
